src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -600,5 +604,32 @@ private static RefKind GetModifiers(SyntaxTokenList modifiers, out SyntaxToken r
 
             return refKind;
         }
+
+        internal static void AddNullCheckingErrorsToParameter(DiagnosticBag diagnostics, ParameterSymbol parameter, bool isLambdaOrLocalFunction)
+        {
+            if (parameter.IsNullChecked)
+            {
+                if (Binder.GetWellKnownTypeMember(parameter.DeclaringCompilation, WellKnownMember.System_ArgumentNullException__ctorString, out DiagnosticInfo diag) is null)
+                {
+                    diagnostics.Add(diag, parameter.Locations.FirstOrNone());
+
+                }
+                if (parameter.Type.IsValueType)
+                {
+                    if (!parameter.Type.IsNullableTypeOrTypeParameter())
+                    {
+                        diagnostics.Add(ErrorCode.ERR_NonNullableValueTypeIsNullChecked, parameter.Locations.FirstOrNone(), parameter);
+                    }
+                    else
+                    {
+                        diagnostics.Add(ErrorCode.WRN_NullCheckingOnNullableValueType, parameter.Locations.FirstOrNone(), parameter);
+                    }
+                }
+            }
+            if (isLambdaOrLocalFunction && parameter.ExplicitDefaultConstantValue?.IsNull == true)
```

Comment:
>isLambdaOrLocalFunction [](start = 16, length = 23)

Is this necessary? Can't `SourceComplexParameterSymbol` use this check?

Reply:
No, it doesn't work unless placed below the conversion in SourceComplexParameterSymbol, and the other ones don't work unless they're placed at the top.